author	Szymon Janc <szymon.janc@tieto.com>
commit	628add0e5c4bdbeaa4ac7c9f87148330dfe2d48d
	Mon, 3 Dec 2012 11:03:41 +0000 (13:03 +0200)

attrib: Fix possible use after free

Move io channels unrefs after last use of channels.

diff -ur bluez-4.99.fix.orig/src/attrib-server.c bluez-4.99.fix/src/attrib-server.c
--- bluez-4.99.fix.orig/src/attrib-server.c	2012-10-17 15:15:25.738254000 -0400
+++ bluez-4.99.fix/src/attrib-server.c	2012-12-10 12:28:23.944253011 -0500
@@ -121,13 +121,13 @@
 	g_list_free_full(server->database, attrib_free);
 
 	if (server->l2cap_io != NULL) {
-		g_io_channel_unref(server->l2cap_io);
 		g_io_channel_shutdown(server->l2cap_io, FALSE, NULL);
+		g_io_channel_unref(server->l2cap_io);
 	}
 
 	if (server->le_io != NULL) {
-		g_io_channel_unref(server->le_io);
 		g_io_channel_shutdown(server->le_io, FALSE, NULL);
+		g_io_channel_unref(server->le_io);
 	}
 
 	g_slist_free_full(server->clients, (GDestroyNotify) channel_free);
